src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -64,6 +64,27 @@ public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContex
                 : StatusCode(StatusCodes.Status409Conflict, "Instance already assigned");
         }
 
+        [HttpPost]
+        [Route("admin/instance/legion-assign")]
```

Comment:
I don't think we need to add a new endpoint for this. Just use the existing admin/instance/assign endpoint. It can handle both scenarios.

Reply:
Are you saying make changes to the existing admin/instance/assign endpoint to be able to accept both encrypted and non-encrypted host assignment context? Currently, the endpoint specifically takes EncryptedHostAssignmentContext  and decrypts it via SetContext() and we have the code changes in websites that will send either the encrypted or the non-encrypted context based on the assign endpoint [code](https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites?path=/src/Hosting/DataService/Microsoft.Web.Hosting.DataService/Runtime/LinuxConsumption/Specialization/Legion/PayloadHelper.cs&version=GBdev&line=141&lineEnd=155&lineStartColumn=13&lineEndColumn=85&lineStyle=plain&_a=contents)

Reply:
Yes, I'm saying that we don't want a new admin/instance/legion-assign endpoint on the host. The existing endpoint suffices and can just vary its behavior based on the incoming payload.

Reply:
Indeed. Thank you for flagging this, @mathewc 

@Tonewall beyond just the name, can we also please make sure we don't have any routes with references specific SKUs or SKU specific logic? We can discuss the design if needed.

Let's also please strive to avoid a lot of SKU specific logic sprinkled throughout the code. As much as possible, let's either consider the existing extensibility points to add support for what you need, or discuss potentially new ones to introduce the logic you need.